### ISSUE TITLE
iCheck('uncheck') not working in own 'ifChecked' callback function

### ISSUE THREAD

**Issue by timmetj** [drgullin/icheck#159]
Hello,

I'm trying to do a simple uncheck of all my other radio buttons when there is one selected.
Also must the radiobuttons be resetted (unchecked) when the main selecter has been checked.

I got 3 different radiogroups with all 1 main activator.
So when i click my main activator, i want the other radiobuttons (from the other 2 groups) to be unchecked, this works.
But when i click on the sub radiobutton in a group, it wont let me uncheck the radiobuttons from another group..

I already tried few method, none work.
 /\* MAIN CODE  (nor working)*/

```
$('input[name="abo"]').on('ifChecked', function() {
    $('.jaar').iCheck('uncheck');
});

$('input.jaar').on('ifChecked', function() {
    $('.jaar').not(this).iCheck('uncheck');
});
```

/*All different groups not working */

```
$('input[name="instap-jaar"], 'input[name="prem-jaar"], 'input[name="super-jaar"]').on('ifChecked', function() {
    $('input[name="prem-jaar"]').iCheck('uncheck');
});
```

/\* Change second function working */

```
$('input[name="instap-jaar"]').on('ifChecked', function() {
    $('input[name="prem-jaar"]').iCheck('uncheck');
});
```

The first function works seperatly, but not together...
The second function just never works, but it does get triggered if you do console.log or so.

It looks like icheck is preventing actions called on the same radiogroup that you call an action..?

**Comment by drgullin** [comment]
Hello @timmetj

Try to use 2.x version.

**Comment by timmetj** [comment]
Sorry, but the 2.x version works even less.. half of my radio buttons don't even check anymore. Only the main radio group gets checked/unchecked, others don't work.
Only code i added atm is the initial code:

```
$('input').icheck({
        checkboxClass: 'icheckbox_square-blue',
        radioClass: 'iradio_square-blue',
        increaseArea: '20%'
});
```

So the iCheck gets loaded, you can see the divs and classes added, also styling is ok, but when i click a radiobutton, it does nothing,  only the frist radio group works. all others on that page dont.

**Comment by somentelucas** [comment]
Same problem here!

```
$('input.label').on('ifClicked', function() {
    $(this).iCheck('unckeck');
});
```

Not working...

**Comment by mfoniso** [comment]
I've got the same issue. Is there a fix yet?
I'm on version v1.0.2 and version two (beta) totally breaks existing functionality.

**Comment by Daijobou** [comment]
In version 2.0 you must use .icheck('ckecked'); // with 'ed' at end

**Comment by filippomortari** [comment]
if you do something like 

``` javascript
 $('.jaar').iCheck('uncheck');
```

you are pointing to the whole radio group. 
In my case, addressing the uncheck method to the specific radio button worked for me. 

``` javascript
$('input:radio[name="foo"]').filter('[value="bar"]').iCheck('uncheck');
```

If you dont know which one is checked you might iterate over every button of the group.

**Comment by francisco2005** [comment]
Solution:
var cual= this;
setTimeout(function(){ $(cual).iCheck('uncheck');}, 1);
or 
setTimeout(function(){ $("#name").iCheck('uncheck');}, 1);

**Comment by neudesk** [comment]
What I did here is:

var checkbox;
checkbox = $('input[name="radio_field"]');
checkbox.iCheck('destroy');
checkbox.prop('checked', false);
$(".icheck-me").each(function() {
  var color, skin;
  skin = "_" + $(this).attr('data-skin');
  color = "-" + $(this).attr('data-color');
  return $(this).iCheck({
    checkboxClass: 'icheckbox' + skin + color,
    radioClass: 'iradio' + skin + color
  });
});

**Comment by congkhuong** [comment]
@francisco2005: Could you please explain for me your solution?

**Comment by Lihatskiy** [comment]
it works)
`
$('input[type="checkbox"]').removeAttr('checked').iCheck('update');
`

**Comment by Woolseyyy** [comment]
@Lihatskiy I don`t know why but it dosen`t work in my case. And I find that the example given ($(input).iCheck('check')) worked. But when I operate on one element, it failed.

**Comment by Woolseyyy** [comment]
oh, 'setTimeout(function(){ $("#name").iCheck('uncheck');}, 1);' works.
thanks @francisco2005 
but why?

**Comment by amar-bhatia** [comment]
sorry for joining late to the party 
use this:
 $('input').on('ifChanged', function(event){
            alert('Unchecked');
            $(this).prop('checked',false).icheck('uncheck');
        }
    });

**Comment by she2** [comment]
This solution by @Lihatskiy only worked a few times then stop working

`$('input[type="checkbox"]').removeAttr('checked').iCheck('update');`

However, @francisco2005 solution works well.

`setTimeout(function(){ $("#name").iCheck('uncheck');}, 1);`

and `setTimeout(function(){ $("#name").iCheck('check');}, 1);`

**Comment by sim6** [comment]
Another workarround for v1.0.2:
```
$('#mycheckbox').iCheck('uncheck').iCheck('update')
```

**Comment by uaoleg** [comment]
> Another workarround for v1.0.2:
> 
> ```
> $('#mycheckbox').iCheck('uncheck').iCheck('update')
> ```

This works, but doesn't fire `ifToggled` event
